packages/server/src/helpers/decodeAuthenticatorExtensions.ts

```diff
@@ -0,0 +1,37 @@
+import cbor from 'cbor';
+
+/**
+ * Convert authenticator extension data buffer to a proper object
+ *
+ * @param extensionData Authenticator Extension Data buffer
+ */
+export default function decodeAuthenticatorExtensionData(
+  extensionData: Buffer
+): AuthenticationExtensionsAuthenticatorOutputs | undefined {
+  let toCBOR: AuthenticationExtensionsAuthenticatorOutputs | undefined;
+  try {
+    toCBOR = cbor.decodeAllSync(extensionData)[0];
+  } catch (err) {
+    const _err = err as Error;
+    throw new Error(`Error decoding authenticator extensions: ${_err.message}`);
+  }
+  return toCBOR;
+}
+
+export type AuthenticationExtensionsAuthenticatorOutputs = {
+  devicePublicKey?: DevicePublicKeyAuthenticatorOutput;
+  uvm?: UvmJSON;
+}
+
+export type DevicePublicKeyAuthenticatorOutput = {
+  dpk?: Buffer;
+  scp?: Buffer;
+  sig?: string;
+  aaguid?: Buffer;
+}
+
+// TODO: Need to verify this format
+// https://w3c.github.io/webauthn/#sctn-uvm-extension.
+export type UvmJSON = {
```

Comment:
Whoops, one more here - after poring over the spec this morning I'm understanding that CDDL definitions are a little handwave-y on guidance of what these types should be named. Can this get renamed to `UVMAuthenticatorOutput` instead please?

```suggestion
  uvm?: UVMAuthenticatorOutput;
}

export type DevicePublicKeyAuthenticatorOutput = {
  dpk?: Buffer;
  scp?: Buffer;
  sig?: string;
  aaguid?: Buffer;
}

// TODO: Need to verify this format
// https://w3c.github.io/webauthn/#sctn-uvm-extension.
export type UVMAuthenticatorOutput = {
```

Reply:
Done.